### PR TITLE
.Net: AzureOpenAI services cleanup

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation, Context, and Description
1. The `ClientCore.DeploymentOrModelName` property is renamed to `ClientCore.DeploymentName` to reflect it's actual purpose.  
2. The `AzureOpenAIPromptExecutionSettings` class is moved to the `Settings` folder to reside alongside prompt execution setting classes for other services.

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs [hunk 1]
```diff
@@ -27,9 +27,9 @@ internal partial class ClientCore
     internal static string DeploymentNameKey => "DeploymentName";
 
     /// <summary>
-    /// Model Id or Deployment Name
+    /// Deployment name or model Id.
     /// </summary>
-    internal string DeploymentOrModelName { get; set; } = string.Empty;
+    internal string DeploymentNameOrModelId { get; set; } = string.Empty;
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs:32]: We should have two seperate properties as these two things are not related e.g. I could have a deploy name as `eastus-gpt-4o` and model id as `gpt-4o`.

Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs]: The property was renamed to `DeploymentName`. Right now, the Azure.AI.OpenAI SDK doesn't let you specify both at the same time and only works with the deployment name. So, there's no need for two properties; the deployment name should be enough for now. 

There are also places in SK where the deployment name is mapped/assigned to the model id. This seems confusing and wrong. This issue captures the problem and will be used to fix it: https://github.com/microsoft/semantic-kernel/issues/7142

**Review by markwallace-microsoft — APPROVED** [review]
